src/BlockTypes/ProductQuery.php

```diff
@@ -37,13 +37,27 @@ protected function initialize() {
 	 * @param array       $parsed_block The block being rendered.
 	 */
 	public function update_query( $pre_render, $parsed_block ) {
-		if ( 'core/query' !== $parsed_block['blockName'] ) {
+		if ( 'core/query' !== $parsed_block['blockName'] || ! isset( $parsed_block['attrs']['__woocommerceVariationProps'] ) ) {
 			return;
 		}
 
 		add_filter(
```

Comment:
I know it's not introduced in this PR, but the `$accepted_args` param of this [`add_filter()`](https://developer.wordpress.org/reference/functions/add_filter/) call could be 1 instead of 3. We are only using `$query`, right?